### PR TITLE
Slack - add filtered_tags parameter

### PR DESCRIPTION
**Pull request by adi88d**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/20175

## Description
Added **filtered_tags** parameter to filter the messages sending from Demisto by tags.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [x] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Slack/Integrations/Slack/Slack.py [hunk 1]
```diff
@@ -1311,8 +1311,20 @@ def slack_send():
     thread_id = demisto.args().get('threadID', '')
     severity = demisto.args().get('severity')  # From server
     blocks = demisto.args().get('blocks')
+    entry_object = demisto.args().get('entryObject')  # From server
     entitlement = ''
 
+    if message_type == MIRROR_TYPE:
```
Comment (liorblob) [inline Packs/Slack/Integrations/Slack/Slack.py:1317]: Place it under the `if` in line 1328

File: Packs/Slack/Integrations/Slack/Slack.py [hunk 2]
```diff
@@ -1311,8 +1311,20 @@ def slack_send():
...
+        tags = argToList(demisto.params().get('filtered_tags', []))
+        entry_tags = entry_object.get('tags', [])
+
+        if tags and not entry_tags:
```
Comment (liorblob) [inline Packs/Slack/Integrations/Slack/Slack.py:1321]: `entry_tags` will never be there in versions under 6.1. We need to explain it properly in the parameter description

File: Packs/Slack/Integrations/Slack/Slack.yml [hunk 3]
```diff
@@ -77,6 +77,11 @@ configuration:
   name: proxy_url
   required: false
   type: 0
+- display: Comma separated list of tags To filter the messages sent from demisto.
```
Comment (liorblob) [inline Packs/Slack/Integrations/Slack/Slack.yml:80]: Only supported in 6.1 and above. @yaakovi How do we want to handle this?

File: Packs/Slack/ReleaseNotes/1_3_8.md [hunk 4]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### Slack v2
+- Added **filtered_tags** parameter to filter the messages sending from Demisto by tags.
```
Comment (liorblob) [inline Packs/Slack/ReleaseNotes/1_3_8.md:4]: Only supported in 6.1 and above.

**Review by liorblob — CHANGES_REQUESTED** [review]
Nice, please add unit tests

File: Packs/Slack/Integrations/Slack/Slack_test.py [hunk 5]
```diff
@@ -4257,3 +4257,55 @@ def test_fail_connect_threads(mocker):
         time.sleep(0.5)
     assert return_error_mock.call_count == 8
     assert threading.active_count() < 6  # we shouldn't have more than 5 threads (1 + 4 max size of executor)
+
+
+def test_slack_send_filter_one_mirro_tag(mocker):
+    # When filtered_tags parameter contains the same tag as the entry tag - slack_send method should send the message
+    import Slack
+    from Slack import slack_send
```
Comment (liorblob) [inline Packs/Slack/Integrations/Slack/Slack_test.py:4265]: no need for this import

File: Packs/Slack/Integrations/Slack/Slack_test.py [hunk 6]
```diff
@@ -4257,3 +4257,55 @@ def test_fail_connect_threads(mocker):
...
+
+    mocker.patch.object(demisto, 'results')
+    mocker.patch.object(demisto, 'getIntegrationContext', side_effect=get_integration_context)
+    mocker.patch.object(Slack, 'slack_send_request', return_value={'cool': 'cool'})
+
+    mocker.patch.object(demisto, 'args', return_value={'to': 'demisto', 'messageType': 'mirrorEntry',
+                                                       'entryObject': {'tags': ['tag1']}})
+
+    mocker.patch.object(demisto, 'params', return_value={'filtered_tags': 'tag1'})
+    slack_send()
```
Comment (liorblob) [inline Packs/Slack/Integrations/Slack/Slack_test.py:4275]:
```suggestion
    Slack.slack_send()
```

File: Packs/Slack/Integrations/Slack/Slack_test.py [hunk 7]
```diff
@@ -4257,3 +4257,55 @@ def test_fail_connect_threads(mocker):
...
+    assert demisto.results.mock_calls[0][1][0]['Contents'] == 'Message sent to Slack successfully.\nThread ID is: None'
+
+
+def test_slack_send_filter_no_mirror_tags(mocker):
+    # When filtered_tags parameter is empty slack_send method should send the message
+    import Slack
+    from Slack import slack_send
```
Comment (liorblob) [inline Packs/Slack/Integrations/Slack/Slack_test.py:4282]: Same

File: Packs/Slack/Integrations/Slack/Slack_test.py [hunk 8]
```diff
@@ -4257,3 +4257,55 @@ def test_fail_connect_threads(mocker):
...
+    mocker.patch.object(demisto, 'params', return_value={'filtered_tags': ''})
+    slack_send()
+    assert demisto.results.mock_calls[0][1][0]['Contents'] == 'Message sent to Slack successfully.\nThread ID is: None'
+
+
+def test_slack_send_filter_no_entry_tags(mocker):
+    # When filtered_tags parameter contains one tag to filter messages and the entry have no tags -
+    # slack_send method should exit and demisto.results.mock_calls should be empty
+    import Slack
+    from Slack import slack_send
```
Comment (liorblob) [inline Packs/Slack/Integrations/Slack/Slack_test.py:4300]: Same

File: Packs/Slack/Integrations/Slack/Slack.yml [hunk 9]
```diff
@@ -77,6 +77,11 @@ configuration:
   name: proxy_url
   required: false
   type: 0
+- display: Comma separated list of tags To filter the messages sent from demisto. Only supported in Demisto V6.1 and above.
```
Comment (liorblob) [inline Packs/Slack/Integrations/Slack/Slack.yml:80]: Might be too much info here, we can split with `additionalInfo`.

**Review by liorblob — APPROVED** [review]
Looks good, minor changes.
Verify descriptions/RN with tech docs

**Review by liorblob — APPROVED** [review]

**Comment by adi88d** [comment]
@amshamah419 please review the `Slack.yml` and `1_3_8.md` files

**Pull request merged**